compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -2297,17 +2297,30 @@ impl<'tcx> TyS<'tcx> {
             | ty::Array(..)
             | ty::Closure(..)
             | ty::Never
-            | ty::Error(_) => true,
+            | ty::Error(_) => Some(true),
 
-            ty::Str | ty::Slice(_) | ty::Dynamic(..) | ty::Foreign(..) => false,
+            ty::Str | ty::Slice(_) | ty::Dynamic(..) | ty::Foreign(..) => Some(false),
 
-            ty::Tuple(tys) => tys.iter().all(|ty| ty.expect_ty().is_trivially_sized(tcx)),
+            ty::Tuple(tys) => {
+                for ty in tys.iter() {
+                    if !ty.expect_ty().is_trivially_sized(tcx)? {
+                        return Some(false);
+                    }
+                }
+                Some(true)
+            }
 
-            ty::Adt(def, _substs) => def.sized_constraint(tcx).is_empty(),
+            ty::Adt(def, _substs) => {
+                if def.sized_constraint(tcx).is_empty() {
+                    Some(true)
+                } else {
+                    None
+                }
+            }
```

Comment:
This would be a lot shorter with https://doc.rust-lang.org/std/primitive.bool.html#method.then_some:
```suggestion
            ty::Adt(def, _substs) => def.sized_constraint(tcx).is_empty().then_some(true),
```

Reply:
Thanks, I didn't know about that method.